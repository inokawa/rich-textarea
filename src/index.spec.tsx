import * as React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Textarea } from ".";

global.ResizeObserver = class mockResizeObjerver {
  instanceResize: ResizeObserver | null = null;
  callbackResize: ResizeObserverCallback | null = null;
  constructor(callback: ResizeObserverCallback) {
    this.instanceResize = this;
    this.callbackResize = callback;
  }
  disconnect() {}
  observe(target: Element, options?: ResizeObserverOptions) {}
  unobserve(target: Element) {}
};

const NOP = () => {};

afterEach(cleanup);

describe("style", () => {
  it("color", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/ore/g, { color: "red" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("decoration", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/ore/g, { textDecoration: "undeline" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("background", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/ore/g, { background: "red" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("border", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/ore/g, { border: "solid 1px red" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("match", () => {
  it("match one", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/ipsum/g, { color: "red" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("match many", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/[or]/g, { color: "red" }]]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("multiple matchers", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[
          [/[or]/g, { color: "red", background: "red" }],
          [/[oe]/g, { color: "blue", border: "solid 1px blue" }],
        ]}
        value={"Lorem ipsum dolor sit amet"}
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("japanese", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[[/[あ-ん]/g, { background: "yellow" }]]}
        value={
          "あのイーハトーヴォのすきとおった風、夏でも底に冷たさをもつ青いそら、うつくしい森で飾られたモリーオ市、郊外のぎらぎらひかる草の波。"
        }
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("emoji", () => {
    const { asFragment } = render(
      <Textarea
        matchers={[
          [
            /[\u{1f300}-\u{1f5ff}\u{1f900}-\u{1f9ff}\u{1f600}-\u{1f64f}\u{1f680}-\u{1f6ff}\u{2600}-\u{26ff}\u{2700}-\u{27bf}\u{1f1e6}-\u{1f1ff}\u{1f191}-\u{1f251}\u{1f004}\u{1f0cf}\u{1f170}-\u{1f171}\u{1f17e}-\u{1f17f}\u{1f18e}\u{3030}\u{2b50}\u{2b55}\u{2934}-\u{2935}\u{2b05}-\u{2b07}\u{2b1b}-\u{2b1c}\u{3297}\u{3299}\u{303d}\u{00a9}\u{00ae}\u{2122}\u{23f3}\u{24c2}\u{23e9}-\u{23ef}\u{25b6}\u{23f8}-\u{23fa}]/gu,
            { background: "yellow" },
          ],
        ]}
        value={
          "Lorem😇 ipsum dolor sit amet👨‍👩‍👧‍👦 Lorem ipsum dolor👍🏽 sit amet Lorem👩‍💻 ipsum dolor sit amet"
        }
        onChange={NOP}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
