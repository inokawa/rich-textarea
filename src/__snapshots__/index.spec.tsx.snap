// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match emoji 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        Lorem
        <span
          style="background: yellow;"
        >
          😇
        </span>
         ipsum dolor sit amet
        <span
          style="background: yellow;"
        >
          👨
        </span>
        ‍
        <span
          style="background: yellow;"
        >
          👩
        </span>
        ‍
        <span
          style="background: yellow;"
        >
          👧
        </span>
        ‍
        <span
          style="background: yellow;"
        >
          👦
        </span>
         Lorem ipsum dolor
        <span
          style="background: yellow;"
        >
          👍
        </span>
        <span
          style="background: yellow;"
        >
          🏽
        </span>
         sit amet Lorem
        <span
          style="background: yellow;"
        >
          👩
        </span>
        ‍
        <span
          style="background: yellow;"
        >
          💻
        </span>
         ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem😇 ipsum dolor sit amet👨‍👩‍👧‍👦 Lorem ipsum dolor👍🏽 sit amet Lorem👩‍💻 ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`match japanese 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        <span
          style="background: yellow;"
        >
          あ
        </span>
        <span
          style="background: yellow;"
        >
          の
        </span>
        イーハトーヴォ
        <span
          style="background: yellow;"
        >
          の
        </span>
        <span
          style="background: yellow;"
        >
          す
        </span>
        <span
          style="background: yellow;"
        >
          き
        </span>
        <span
          style="background: yellow;"
        >
          と
        </span>
        <span
          style="background: yellow;"
        >
          お
        </span>
        <span
          style="background: yellow;"
        >
          っ
        </span>
        <span
          style="background: yellow;"
        >
          た
        </span>
        風、夏
        <span
          style="background: yellow;"
        >
          で
        </span>
        <span
          style="background: yellow;"
        >
          も
        </span>
        底
        <span
          style="background: yellow;"
        >
          に
        </span>
        冷
        <span
          style="background: yellow;"
        >
          た
        </span>
        <span
          style="background: yellow;"
        >
          さ
        </span>
        <span
          style="background: yellow;"
        >
          を
        </span>
        <span
          style="background: yellow;"
        >
          も
        </span>
        <span
          style="background: yellow;"
        >
          つ
        </span>
        青
        <span
          style="background: yellow;"
        >
          い
        </span>
        <span
          style="background: yellow;"
        >
          そ
        </span>
        <span
          style="background: yellow;"
        >
          ら
        </span>
        、
        <span
          style="background: yellow;"
        >
          う
        </span>
        <span
          style="background: yellow;"
        >
          つ
        </span>
        <span
          style="background: yellow;"
        >
          く
        </span>
        <span
          style="background: yellow;"
        >
          し
        </span>
        <span
          style="background: yellow;"
        >
          い
        </span>
        森
        <span
          style="background: yellow;"
        >
          で
        </span>
        飾
        <span
          style="background: yellow;"
        >
          ら
        </span>
        <span
          style="background: yellow;"
        >
          れ
        </span>
        <span
          style="background: yellow;"
        >
          た
        </span>
        モリーオ市、郊外
        <span
          style="background: yellow;"
        >
          の
        </span>
        <span
          style="background: yellow;"
        >
          ぎ
        </span>
        <span
          style="background: yellow;"
        >
          ら
        </span>
        <span
          style="background: yellow;"
        >
          ぎ
        </span>
        <span
          style="background: yellow;"
        >
          ら
        </span>
        <span
          style="background: yellow;"
        >
          ひ
        </span>
        <span
          style="background: yellow;"
        >
          か
        </span>
        <span
          style="background: yellow;"
        >
          る
        </span>
        草
        <span
          style="background: yellow;"
        >
          の
        </span>
        波。
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      あのイーハトーヴォのすきとおった風、夏でも底に冷たさをもつ青いそら、うつくしい森で飾られたモリーオ市、郊外のぎらぎらひかる草の波。
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`match match many 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          o
        </span>
        <span
          style="color: red;"
        >
          r
        </span>
        em ipsum d
        <span
          style="color: red;"
        >
          o
        </span>
        l
        <span
          style="color: red;"
        >
          o
        </span>
        <span
          style="color: red;"
        >
          r
        </span>
         sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`match match one 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        Lorem 
        <span
          style="color: red;"
        >
          ipsum
        </span>
         dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`match multiple matchers 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: blue; background: red; border: 1px solid blue;"
        >
          o
        </span>
        <span
          style="color: red; background: red;"
        >
          r
        </span>
        <span
          style="color: blue; border: 1px solid blue;"
        >
          e
        </span>
        m ipsum d
        <span
          style="color: blue; background: red; border: 1px solid blue;"
        >
          o
        </span>
        l
        <span
          style="color: blue; background: red; border: 1px solid blue;"
        >
          o
        </span>
        <span
          style="color: red; background: red;"
        >
          r
        </span>
         sit am
        <span
          style="color: blue; border: 1px solid blue;"
        >
          e
        </span>
        t
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea background 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px; background: blue;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea background-color 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px; background-color: blue;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea border 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid blue; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="border: 1px solid blue; background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea border-color 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="border-color: blue; background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea caret-color 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="caret-color: blue; background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea color 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="color: transparent; background: transparent; margin: 0px; position: absolute; caret-color: blue;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style textarea width & height 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="width: 123px; height: 456px; background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style value background 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="background: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style value border 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="border: 1px solid red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style value color 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="color: red;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;

exports[`style value decoration 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; z-index: -1; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; transform: translate(0px, 0px); pointer-events: none; user-select: none; padding: 2px 2px 2px 2px; margin: 0px 0px 0px 0px; border: 1px solid; border-width: 1px; border-top-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-right-width: 1px; border-style: solid; border-top-style: solid; border-bottom-style: solid; border-left-style: solid; border-right-style: solid; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; white-space: pre-wrap; border-color: transparent;"
      >
        L
        <span
          style="text-decoration: undeline;"
        >
          ore
        </span>
        m ipsum dolor sit amet
      </div>
    </div>
    <textarea
      style="background: transparent; margin: 0px; position: absolute; color: transparent;"
    >
      Lorem ipsum dolor sit amet
    </textarea>
  </div>
</DocumentFragment>
`;
